### PR TITLE
Adds promise-based API for asynchronous ViziCities methods

### PR DESCRIPTION
**Pull request by robhawkes**
# Description

ViziCities methods are currently not using promises, meaning that asynchronous logic is either handled using callbacks or it run after the return of a method, providing no means of knowing when execution has completed.

# Solution

Most asynchronous and potentially asynchronous methods in ViziCities now return a promise that can be used to defer further logic until async execution has completed.

# Limitations

This change means that some ViziCities methods now return a `Promise` object rather than `this` referring to the scope of the method (eg. for chaining). All existing examples work without modification though it's possible that some uses of ViziCities may break because of the change in what is returned.

### CODE REVIEW

**Comment by robhawkes** [comment]
Had a bit of a nightmare rebasing this with the changes within the dev branch. Pretty sure it's all up to date now.

File: src/controls/Controls.Orbit.js [hunk 1]
```diff
@@ -206,7 +209,7 @@ class Orbit extends EventEmitter {
   // Add controls to world instance and store world reference
   addTo(world) {
     world.addControls(this);
```
Comment (robhawkes) [inline src/controls/Controls.Orbit.js]: Return `world.addControls` instead of creating a new promise

File: src/layer/GeoJSONLayer.js [hunk 2]
```diff
@@ -62,27 +62,31 @@ class GeoJSONLayer extends LayerGroup {
...
+        url: url,
+        type: 'json',
+        crossOrigin: true
+      }).then(res => {
+        // Clear request reference
+        this._request = null;
+        this._processData(res).then(() => {
+          resolve(this);
+        });
+      }).catch(err => {
```
Comment (robhawkes) [inline src/layer/GeoJSONLayer.js]: Call `reject` in here

**Comment by matthojo** [comment]
Looks good 👍🏻

**Pull request merged**